File: Sorting/insertion_sort/algo-1/1.py
# Implementing Insertion Sort in Pyhton using Basic Synatx
# Input the Size Of Array
N = int(input())
# Declare a empty List to store elements
L = []
# To read the elements
for i in range(N):
     # Adding element in list using list.append()
     L.append(int(input()))

# Insertion Sort Algorithm
# we can also use numpy.arange()
for i in range(N-1):
    key = L[i+1]
    j = i
    while (j >= 0 and L[j] > key):
        L[j+1] = L[j]
        j=j-1
    L[j+1] = key

# Output
for i in range(N):
    print(L[i],end=" ")

File: Sorting/insertion_sort/algo-1/1.php
<?php
	// Insertion Sort Using php 
	// Read Input from standard input file 
	$handle = fopen ("php://stdin","r");
	// Read Size Of Array
	$N = (int)fgets($handle);
	$A = array();
	// Input Array elements
	for($i=0;$i<$N;$i++){
		// We can also use array_push($A,$number)
		$A[$i] = (int)fgets($handle);	
	}
	// Insertion Sort Algorithm
	for($i=1;$i<$N;$i++){
		$key = $A[$i];
		$j = $i - 1;
		while($j>=0 && $A[$j]>$key){
			$A[$j+1] = $A[$j];	
			$j--;
		}
		$A[$j+1] = $key;
	}
	// Output
	for($i=0;$i<$N;$i++){
		echo $A[$i]." ";
	}
?>
